include/vara/Feature/Constraint.h

```diff
@@ -481,10 +587,14 @@ class Feature;
 class PrimaryFeatureConstraint : public PrimaryConstraint {
 public:
   PrimaryFeatureConstraint(std::variant<Feature *, std::unique_ptr<Feature>> FV)
-      : PrimaryConstraint(ConstraintKind::CK_FEATURE), FV(std::move(FV))  {}
+      : PrimaryConstraint(ConstraintKind::CK_FEATURE), FV(std::move(FV)) {}
 
   [[nodiscard]] Feature *getFeature() const;
 
+  std::unique_ptr<Constraint> clone() override {
+    return std::make_unique<PrimaryFeatureConstraint>(this->getFeature());
```

Comment:
Is this feature ptr later replaced?

Reply:
I reworked this part, so that this copy only holds a new shallow feature (name only), which later needs replacement (the `FeatureModelBuilder` used cloning a `FeatureModel` already does exactly that).

Also, added comment should explain the intended behavior.